C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
+    - match: $
+      pop: true
+
+  option_message:
+    - meta_content_scope: string.unquoted.cs
+    - include: comments
+    - match: $
+      pop: true
+
+  # option_done pops out at the end of the line and handles comments
+  option_done:
+    - include: comments
+    - match: $
+      pop: true
+    - match: \S
+      scope: invalid.illegal.cs
+
+  main:
+    - match: '\}'
+      scope: invalid.illegal.unexpected.closing-bracket
+
+    - match: '\{'
+      scope: punctuation.section.block.begin.cs
+      push:
+        - match: '\}'
+          scope: punctuation.section.block.end.cs
+          pop: true
+        - include: main
+
+    # skips opening
+    - match: ^\s*
+
+    - include: tag
+    - include: using
+    - include: class_declaration
+    - include: interface_declaration
+    - include: namespace_declaration
+    - include: delegate_declaration
+    # - include: method_declaration
+
+    # allows coloration of code outside a class
+    - include: code_block_in
+
+  using:
+    - match: '\b(using)\s+{{name}}\s*(=)\s*'
+      captures:
+        1: keyword.control.import.cs
+        2: entity.name.type.cs
+        3: storage.modifier.namespace.as.cs
+      push: using_namespace
+    - match: '\b(using)\s+(static\b)?\s*{{name}}'
+      captures:
+        1: keyword.control.import.cs
+        2: storage.modifier.namespace.cs
+        3: variable.namespace.cs
+      push: using_namespace
+
+  using_namespace:
+    - match: '{{name}}'
+      scope: variable.namespace.cs
+    - match: \.
+      scope: punctuation.separator.namespace.cs
+    - match: '<'
+      scope: punctuation.definition.arguments.type.begin.cs
+      push: type_argument
+    - match: ';'
+      scope: punctuation.definition.namespace.end.cs
+      pop: true
+    - match: '.*$'
+      scope: invalid.illegal.expected-namespace.cs
+      pop: true
+
+  namespace_declaration:
+    # package declaration
+    - match: '\b(namespace)\b'
+      captures:
+        0: meta.package.identifier.cs
+        1: storage.type.namespace
+        2: entity.name.namespace.cs
+        3: punctuation.accessor.dot.cs
+        4: entity.name.namespace.cs
+      push:
+        - match: '{{name}}'
+          scope: entity.name.namespace.cs
+        - match: \.
+          scope: punctuation.separator.namespace.cs
+        - match: \{
+          scope: punctuation.section.namespace.begin.cs
+          set:
+            - meta_scope: meta.namespace.cs
+            - match: \}
+              scope: punctuation.section.namespace.end.cs
               pop: true
-            - include: method
-            - match: '='
-              scope: keyword.operator.assignment.cs
-              push:
+            - include: main
+        - match: (?=[^\s{])
+          pop: true
+
+  class_declaration:
+    - match: '\b(static|unsafe|abstract|partial|sealed)\b'
+      scope: storage.modifier.cs
+    - match: '{{visibility}}'
+      scope: storage.modifier.access.cs
+    - match: '\b(class)\s+{{name}}'
```

Comment:
There should be a fallback somewhere (there may already be) for base `class` to pick it up before someone has typed the space or the name after it.